ospfd/ospf_interface.c

```diff
@@ -845,18 +845,20 @@ int ospf_if_down(struct ospf_interface *oi)
 	/* Shutdown packet reception and sending */
 	ospf_if_stream_unset(oi);
 
-	for (rn = route_top(ospf->new_table); rn; rn = route_next(rn)) {
-		or = rn->info;
+	if (ospf->new_table) {
```

Comment:
```if (!ospf->new_table)
           return
```

Would look better and have a lot less code change.

Reply:
Can this be fixed as @donaldsharp suggested?